src/app/utilities/ui-helper.ts

```diff
@@ -24,4 +26,23 @@ export class UIHelper {
       }
     }).bind(this), delay);
   }
+
+  /**
+   * The PrimeNg library does not handle selecting of a parent node
+   * when selecting the node programmatically
+   * @param node parent of which has to be (partially) selected
+   * @param selectedNodes array of currently selecte nodes
+   */
+  public static selectPrimeNgParentTreeNode(node: TreeNode, selectedNodes: GbTreeNode[]) {
+    if (node.parent != null) {
```

Comment:
If it would be `selectPrimeNgTreeNodeAndAncestors(node: TreeNode, selectedNodes: GbTreeNode[])`, the structure would be more like:
```typescript
if (node === null) {
  return;
}
selectedNodes.push(node);
if (!node.children.every((child: TreeNode) => selectedNodes.includes(child))) {
  node.partialSelected = true;
}
selectPrimeNgTreeNodeAndAncestors(node.parent);
```

Reply:
Right?

Reply:
My main concern is that this function only uses `node.parent` and never `node` itself. You might as well pass `node.parent` to the function then.

Reply:
```typescript
public static function recursivelySelectParentNodes(parent: TreeNode, selectedNodes: GBTreeNode[])`
  if (parent === null) {
    return;
  }
  if (parent.children.every((child: TreeNode) => selectedNodes.includes(child))) {
    selectedNodes.push(parent);
  } else {
    parent.partialSelected = true;
  }
  selectPrimeNgTreeNodeAndAncestors(parent.parent);
}
```

Reply:
thanks for the comments, I'll update this